influxql/ast.go

```diff
@@ -1133,6 +1133,23 @@ func (s *SelectStatement) validateAggregates(tr targetRequirement) error {
 				if err := s.validSelectWithAggregate(numAggregates); err != nil {
 					return err
 				}
+				// Validate that if they have a time dimension, they need a sub-call like min/max, etc.
+				if s.hasTimeDimensions(s.Condition) {
+					var valid bool
+					for _, a := range expr.Args {
+						if c, ok := a.(*Call); ok {
+							n := strings.ToLower(c.Name)
+							if n == "min" || n == "max" || n == "first" || n == "last" || n == "mean" || n == "sum" {
```

Comment:
That's a good question.  I felt the same way, but @pauldix specifically wanted the error to list them, so that was automatically brittle.  It made sense that the check as identical to the error.  I think either we have it in both spots, or take it out of both spots.